server/api/launcher.py

```diff
@@ -268,6 +278,15 @@ def _validate_runtime(
             )
 
         self._validate_state_thresholds(run.spec.state_thresholds)
+
+        if (
+            mlrun.runtimes.RuntimeKinds.requires_image_name_for_execution(runtime.kind)
+            and not runtime.spec.image
```

Comment:
why not use `is_deployed`? because of spark?

Reply:
Yes, Spark is behaving differently and has a `use_default_image` field in the `Spark3JobSpec` class. That's also why `requires_image_name_for_execution` doesn't just look for local kinds but also for Spark.